installer.sh

```diff
@@ -0,0 +1,423 @@
+        "comment")
+            style_start="\033[2m"
+            style_end="\033[22;39m"
+            ;;
+    esac
+    fi
+
+    builtin echo -e "${style_start}${1}${style_end}"
+}
+
+function exit_with_error() {
+    output "+--------------------------------------------------+" "error"
+    output "|                                                  |" "error"
+    output "|              Installation failed                 |" "error"
+    output "|                                                  |" "error"
+    output "+--------------------------------------------------+" "error"
+
+    output "\nGet help with your Platform.sh installation:" "heading"
+    output "  Inspect the logs: ${INSTALL_LOG}"
+    output "  Website:  https://platform.sh"
+
+    exit 1
+}
+
+function intro() {
+    output "+--------------------------------------------------+" "heading"
+    output "|                                                  |" "heading"
+    output "|             Platform.sh Installer                |" "heading"
+    output "|                                                  |" "heading"
+    output "+--------------------------------------------------+" "heading"
+
+    output "\nChecking Environment" "heading"
+}
+
+function outro() {
+    output ""
+    output "+--------------------------------------------------+" "success"
+    output "|                                                  |" "success"
+    output "|   Platform.sh has been installed successfully.   |" "success"
+    output "|                                                  |" "success"
+    output "+--------------------------------------------------+" "success"
+
+    output "\nWhat's next?" "heading"
+
+    output "\nUseful links:" "heading"
+    output "  Official website: https://platform.sh"
+    output "  Documentation:    https://docs.platform.sh"
+    output "  CLI introduction: https://docs.platform.sh/get-started/introduction.html#cli"
+
+    if [ ! -z "$footer_notes" ]; then
+        output "\nWarning during installation:" "heading"
+        output "$footer_notes" "warning"
+    fi
+
+    output "\nThank you for using Platform.sh!"
+}
+
+function add_footer_note() {
+    for var in "$@"; do
+        if [ ! -z "$footer_notes" ]; then
+            footer_notes="${footer_notes}\n${var}"
+        else
+            footer_notes="${footer_notes}${var}"
+        fi
+    done
+}
+
+function indent() {
+    OLDIFS=$IFS
+    IFS=$'\n'
+    while read -r data; do
+        line=$(echo "   | ${data}"|sed $'s/\r/\r   | /g'|sed $'s/\x1B\[[0-9;]\{1,\}[A-Za-z]//g')
+        output "$line" "comment"
+    done
+    IFS=$OLDIFS
+}
+
+# Check that cURL is installed
+function check_curl() {
+    if command -v curl >/dev/null 2>&1; then
+        output "  [*] cURL is installed" "success"
+        cmd_download="curl --silent --fail --location"
+        if [ ! -z $GITHUB_TOKEN ]; then
+            cmd_download="curl --silent --fail --location --header \"authorization: Bearer $GITHUB_TOKEN\""
```

Comment:
Is this definitely `Bearer` not `token`?

(looks like the GitHub docs have changed drastically since they introduced their own CLI)

Reply:
there's a mysterious "In most cases" on https://docs.github.com/en/developers/apps/building-github-apps/authenticating-with-github-apps#authenticating-as-an-installation 
![image](https://user-images.githubusercontent.com/1465106/215114261-e24dde5e-21e1-48e5-af65-c8541b3597c3.png)

Reply:
Maybe `token` is a BC thing, I've always used `Bearer` (or at least that's what I remember) and `Bearer` is the standard one.

Also, from this note I understand `Bearer` will always work, while `token` might not work when passing a JWT.